parts/kuberneteswindowssetup.ps1

```diff
@@ -0,0 +1,299 @@
+<#
+    .SYNOPSIS
+        Provisions VM as a Kubernetes agent.
+
+    .DESCRIPTION
+        Provisions VM as a Kubernetes agent.
+#>
+[CmdletBinding(DefaultParameterSetName="Standard")]
+param(
+    [string]
+    [ValidateNotNullOrEmpty()]
+    $MasterIP = "10.240.255.5",
+
+    [parameter()]
+    [ValidateNotNullOrEmpty()]
+    $KubeDnsServiceIp = "10.0.0.10",
+
+    [parameter()]
+    [ValidateNotNullOrEmpty()]
+    $CACertificate = "<<<caCertificate>>>",

```

Comment:
Why this dual parameterization? Either:

1. The script takes these as args, we should just pass them as args.

2. The script is only used by us and we interpolate directly in. In this case, don't make them parameters, and make the other variables work the same.